### DISCUSSION TITLE
Using 3rd party libraries in Theia

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#10433] [Q&A]
Is it possible to embed a 3rd party element such as an Angular Element within a Theia extension?  I posted this question during TheiaCon 2021, but the sound dropped off before the question could be asked.  We have a large angular application that took a long time to develop.  We have refactored it so that we can convert smaller pieces of it into Angular Elements.  One of them would be extremely difficult to rewrite in React, so we were hoping we could somehow embed it within a Theia extension that we are developing.

**Comment by vince-fugnitto** [comment]
@BarryDrez thank you for the discussion and for attending the conference :+1:

As for your question, I do not believe the framework prevents the use of third party UI frameworks from being used, so it should definitely be possible. At the moment the default widgets of the framework use React, but for new components which you'll write I believe you can do so with Angular if you require so. Unfortunately, I'm not sure how tricky it is to get everything set up, and I don't believe there is any official example at the time of this writing.

**Reply by BarryDrez** [reply]
Thank you @vince-fugnitto.  I am new to React and Theia, so I really have no idea how to include a third party UI within a Theia extension.  I did a lot of digging and you're right - I couldn't find any examples.
